scope/xgb.py

```diff
@@ -0,0 +1,206 @@
+import pandas as pd
+
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+import os
+import sys
+import copy
+import json
+
+
+class XGB(AbstractClassifier):
+    """Baseline model with a statically-defined architecture"""
+
+    def setup(
+        self,
+        **kwargs,
+    )
+
+    #I need to create a call for the run's specification:
+    ## ie xgbrun = {'datasetname': 'd15',
+      #      'classifiername': 'yso',
+       #     'featuresetname': 'f'}
+
+#modelprefix = xgbrun['datasetname'] +'.'+ xgbrun['classifiername'] +'.'+ xgbrun['featuresetname']
+#modelname = modelprefix + '.model'
+#fpfile = modelprefix + '_FPs.csv'
+#fnfile = modelprefix + '_FNs.csv'
+#density = modelprefix + '_density.png'
+#cmpng = modelprefix + '_cm.png'
+#histpng = modelprefix + '_hist.png'
+#impvars = modelprefix + '_impvars.png'
+#fnrfpr = modelprefix + '_fnrfpr.png'
+#recallpng = modelprefix + '_recall.png'
+#rocpng = modelprefix + '_roc.png'
+#jsonname = modelprefix + '.json'
+#txtfile = modelprefix + '.txt'
+#f = open(txtfile, "w")
+#f.write("dataset: %s\nclassifier: %s\nfeatureset: %s\n" % (xgbrun['datasetname'], xgbrun['classifiername'], xgbrun['featuresetname']))
+# LEAVE THIS UNTIL THE END WHEN RUNNING, so all data goes into file f: f.close()
+
+
+#READ IN DATASET
+
+#Choose features/labels/ids, seed/test size/train-test-split
+
+#Define params necessary for a training::
+params = {
+    # Parameters that we are going to tun? These are hyperparameters, so we only tune them with cross-validation rounds...
+    'max_depth':6,
+    'min_child_weight': 1,
+    'eta':.1,
+    'subsample': 0.7,
+    'colsample_bytree': 0.7,
+    # Other parameters
+#    'objective':'reg:squarederror',   # changed from reg:linear
+    'objective': 'binary:logistic',
+  #  'gpu_id': 0,
+  #  'tree_method': 'gpu_hist',
+    'eval_metric': 'auc'
+}
+    def train(
+        params,
+        dtrain,
+        num_boost_round=num_boost_round,
+        evals=[(dtest, "Test")],
+        early_stopping_rounds=10
+)
+    
+    def evaluate(self, test_dataset, **kwargs):
+        return self.model.evaluate(test_dataset, **kwargs)
+
+    def predict(self, x, **kwargs):
+        return self.model.predict(x, **kwargs)
+
+    
+    def load(self, path_model, **kwargs):
+        self.model.load_weights(path_model, **kwargs)
+
+    def save(
+        self,
+        tag: str,
+        output_path: str = "./",
+        output_format: str = "h5",
+    ):
+
+        if output_format not in ("h5",):
+            raise ValueError("unknown output format")
+
+        path = pathlib.Path(output_path)
+        if not path.exists():
+            path.mkdir(parents=True, exist_ok=True)
+
+        output_name = self.name if not tag else f"{self.name}.{tag}"
+        if not output_name.endswith('.h5'):
+            output_name += '.h5'
+        self.model.save(path / output_name, save_format=output_format)
+
+    def hist_rates(self,preds):
```

Comment:
While the plotting will be a useful diagnostic tool going forward with this algorithm, we can first focus on getting the tools associated with training and inference running smoothly.